scene/gui/spin_box.cpp

```diff
@@ -181,8 +185,14 @@ void SpinBox::gui_input(const Ref<InputEvent> &p_event) {
 	}
 }
 
+void SpinBox::_line_edit_focus_enter() {
+	int col = line_edit->get_caret_column();
+	_value_changed(0); // Update the LineEdit's text.
```

Comment:
Won't this force resetting the text to `0` whenever you focus it? That doesn't sound like expected behavior.

Reply:
Argument of the `_value_changed()` is not the new value, but how much the value has changed. So it doesn't change the value here.

Reply:
Are you sure? The code does read like it will set the LineEdit to 0.
```

Reply:
Oh it's `value` and doesn't use `p_value` at all aside from passing it to Range. Super confusing.

Reply:
Well, it works, soo... 🤷‍♂️
![godot windows tools x86_64_KrarW3nuYt](https://user-images.githubusercontent.com/2223172/186889864-d6ae2b13-79ee-4788-a1ba-e7ad787b724a.gif)